client/src/js/addendum/AddendumForm.jsx

```diff
@@ -0,0 +1,284 @@
+  const [canManage, setCanManage] = useState(false)
+
+  React.useEffect(() => {
+    props.agreement.getAddendums(props.addendumType)
+      .then(setAddendums)
+      .catch(console.error)
+  }, [props.agreement])
+
+  React.useEffect(() => {
+    props.agreement.getWhitelist(props.addendumType)
+      .then((res) => {
+        setActiveIdentities(res)
+      })
+  }, [props.agreement, addendums])
+
+  React.useEffect(() => {
+    // FIXME this seems to be state more than effect, doublecheck
+    if (addendums && addendums.length) {
+      setLastAddendum(addendums[addendums.length - 1])
+    } else {
+      setLastAddendum(null)
+    }
+  }, [addendums])
+
+  React.useEffect(() => {
+    (async () => {
+      AddendumFormCtrl.isOwnerOrManager(props.user, props.agreement)
+        .then(status => {
+          setCanManage(status)
+        })
+    })()
+  }, [props.agreement])
+
+  const createAddendum = () => {
+
+    const signer = new Identity(
+      IdentityType.EMAIL,
+      props.user.displayName, // FIXME displayName may not be set, how do we get the user name?
+      props.user.email
+    )
+
+    const addendum = new Addendum(
+      props.addendumType,
+      props.agreement.id,
+      signer,
+      addedIdentities,
+      removedIdentities
+    )
+
+    markUpdateStatus(true)
+    addendum.save()
+      .then(res => {
+        setAddedIdentities([])
+        setRemovedIdentities([])
+      })
+      .catch(err => {
+        if (err.code === 'permission-denied') {
+          setError('Permission denied, please try again later')
+          return
+        }
+        setError('Request failed, please try again later')
+      })
+      .finally(() => {
+        markUpdateStatus(false)
+        setAddendums(addendums => [...addendums, addendum])
+      })
+  }
+
+  const setAddedIdentity = (identity) => {
+    setAddedIdentities(addedIdentities => [identity, ...addedIdentities])
+    markUpdateStatus(true)
+  }
+
+  const setRemovedIdentity = (identity) => {
+    return (evt) => {
+      evt.preventDefault()
+      setRemovedIdentities(removedIdentities => [identity, ...removedIdentities])
+      _.remove(activeIdentities, identity)
+      setActiveIdentities(activeIdentities)
+      markUpdateStatus(true)
+    }
+  }
+
+  const undoRemove = (identity) => {
+    return (evt) => {
+      evt.preventDefault()
+      _.remove(removedIdentities, identity)
+      setRemovedIdentities(removedIdentities)
+      setActiveIdentities(activeIdentities => [identity, ...activeIdentities])
+    }
+  }
+
+  const undoAdd = (identity) => {
+    return (evt) => {
+      evt.preventDefault()
+      setAddedIdentities(_.without(addedIdentities, identity))
+    }
+  }
+
+  const markUpdateStatus = (value) => {
+    props.updateStatus(value)
+  }
+
+  const headers = {}
+  headers[AddendumType.CONTRIBUTOR] = 'Active Contributors for this Agreement'
+  headers[AddendumType.COSIGNER] = 'Active Managers for this Agreement'
```

Comment:
nit: use terminology consistently throughout the code. I remember discussing that "COSIGNER" is not the appropriate legal term. I'm not saying that code has legal value, but I would use MANAGER consistently to improve readability.
```suggestion
  headers[AddendumType.MANAGER] = 'Active Managers for this Agreement'
```